build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/BuildDockerCreate.java

```diff
@@ -60,15 +75,27 @@ public static void main(String[] ignored) {
             // Client builders.
             ArtifactoryManagerBuilder artifactoryManagerBuilder = new ArtifactoryManagerBuilder().setClientConfiguration(clientConfiguration, clientConfiguration.publisher);
             // Load artifact and BuildInfo properties from publisher section in the BuildInfo.properties file.
-            ArtifactoryClientConfiguration.KanikoHandler kanikoHandler = clientConfiguration.kanikoHandler;
+            ArtifactoryClientConfiguration.DockerHandler dockerHandler = clientConfiguration.dockerHandler;
+            String imageFile;
+            ImageFileType imageFileType;
+            if (StringUtils.isNotBlank(dockerHandler.getKanikoImageFile())) {
+                imageFile = dockerHandler.getKanikoImageFile();
+                imageFileType = ImageFileType.KANIKO;
+            } else if (StringUtils.isNotBlank(dockerHandler.getJibImageFile())) {
+                imageFile = dockerHandler.getJibImageFile();
+                imageFileType = ImageFileType.JIB;
+            } else {
+                throw new RuntimeException("kaniko.image.file or jib.image.file property is expected");
+            }
             // Init BuildDockerCreate.
             BuildDockerCreate dockerBuildCreate = new BuildDockerCreate(artifactoryManagerBuilder,
-                    kanikoHandler.getImageFile(),
+                    imageFile,
+                    imageFileType,
                     ArrayListMultimap.create(clientConfiguration.publisher.getMatrixParams().asMultimap()),
                     clientConfiguration.publisher.getRepoKey(),
                     clientConfiguration.getLog());
 
-            // Exe docker push & collect build info.
+            // Exe build-docker-create & collect build info.
             dockerBuildCreate.executeAndSaveBuildInfo(clientConfiguration);
```

Comment:
executeAndSaveBuildInfo --> createAndPublish

Reply:
The `executeAndSaveBuildInfo` is a method of `PackageManagerExtractor`, the parent class.